### PR TITLE
NxSmallThreatCounter RTL Conversion - RSC-1362

### PR DESCRIPTION
**Pull request by cindyko226**
https://issues.sonatype.org/browse/RSC-1362

### CODE REVIEW

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx [hunk 1]
```diff
@@ -4,280 +4,157 @@
...
 
 import NxSmallThreatCounter, { Props } from '../NxSmallThreatCounter';
 
 describe('NxSmallThreatCounter', function() {
-  const minimalProps = {},
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxSmallThreatCounter, minimalProps);
+  const quickRender = rtlRender<Props>(NxSmallThreatCounter, {}),
+      renderEl = rtlRenderElement<Props>(NxSmallThreatCounter, {});
+
+  function getCounter(component: RenderResult, category: string) {
```
Comment (rpokorny) [inline lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx:17]: Probably best to use `getByTitle` for these

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx [hunk 2]
```diff
@@ -4,280 +4,157 @@
...
-    expect(moderateCount.find('.nx-small-threat-counter__category')).toHaveText('Moderate');
-    expect(lowCount.find('.nx-small-threat-counter__count')).toHaveText('33');
-    expect(lowCount.find('.nx-small-threat-counter__category')).toHaveText('Low');
-    expect(noneCount.find('.nx-small-threat-counter__count')).toHaveText('22');
-    expect(noneCount.find('.nx-small-threat-counter__category')).toHaveText('None');
-    expect(unspecifiedCount.find('.nx-small-threat-counter__count')).toHaveText('42');
-    expect(unspecifiedCount.find('.nx-small-threat-counter__category')).toHaveText('Unspecified');
+    expect(counter).toBeInTheDocument();
+    expect(counter).toHaveTextContent('0');
+    await waitFor(() => expect(counter).toHaveAccessibleName('Critical'));
```
Comment (rpokorny) [inline lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx:63]: If you call `await runTimers()` before the expectation I don't think you'll need the `waitFor`

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx [hunk 3]
```diff
@@ -4,280 +4,157 @@
...
+    expect(moderateCounter).toHaveTextContent('44');
+    expect(lowCounter).toHaveTextContent('33');
+    expect(noneCounter).toHaveTextContent('22');
+    expect(unspecifiedCounter).toHaveTextContent('11');
+  });
+
+  it('displays maxDigits\' nines and a plus sign for any counts that go over the max when maxDigits is specified' +
+    'default is three digits', function() {
+    const twoDigitMaxComponent = quickRender({ criticalCount: 100000, maxDigits: 2 }),
+        fiveDigitDefaultComponent = quickRender({ moderateCount: 1000000 }),
```
Comment (rpokorny) [inline lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx:138]: five?

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx [hunk 4]
```diff
@@ -4,280 +4,157 @@
...
+
+    expect(criticalCounter).toHaveTextContent('66');
+    expect(severeCounter).toHaveTextContent('55');
+    expect(moderateCounter).toHaveTextContent('44');
+    expect(lowCounter).toHaveTextContent('33');
+    expect(noneCounter).toHaveTextContent('22');
+    expect(unspecifiedCounter).toHaveTextContent('11');
+  });
+
+  it('displays maxDigits\' nines and a plus sign for any counts that go over the max when maxDigits is specified' +
```
Comment (rpokorny) [inline lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx:135]:
```suggestion
  it('displays maxDigits\' nines and a plus sign for any counts that go over the max when maxDigits is specified. ' +
```

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx (hunk 1 above)
Comment (cindyko226) [reply lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx]: Agree, it has the title attribute.
Updated here 90384b82c1b3e088c4144729e663709de552c392, thanks.

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx (hunk 2 above)
Comment (cindyko226) [reply lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx]: Updated here 90384b82c1b3e088c4144729e663709de552c392

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx (hunk 3 above)
Comment (cindyko226) [reply lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx]: Corrected here 1134bd59d82384c4fa802b214beaf740d78e7307.

Comment (rpokorny) [reply lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx]: > Corrected here https://github.com/sonatype/sonatype-react-shared-components/commit/1134bd59d82384c4fa802b214beaf740d78e7307.

That wasn't the change I was expecting.  Based on the variable name above this one, I thought the number in the variable name referred to the `maxDigits` value.  So shouldn't this one be `threeDigitDefaultComponent`?

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx [hunk 5]
```diff
@@ -4,280 +4,176 @@
...
-    expect(unspecifiedCount.find('.nx-small-threat-counter__category')).toHaveText('Unspecified');
-  });
+    const counter = component.getByTitle('Critical');
 
-  it('correctly assigns supplied class', function() {
-    const classComponent = getMountedComponent({ criticalCount: 0, className: 'test-class' });
-    expect(classComponent.find('.nx-small-threat-counter-container')).toHaveClassName('test-class');
+    expect(counter).toBeInTheDocument();
+    expect(counter).toHaveTextContent('0');
+    expect(counter).toHaveAccessibleName('Critical');
```
Comment (rpokorny) [inline lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx:58]: I actually don't think we need to test the accessible name here.  `<div>`s have a role of `generic`, and the `generic` role prohibits naming.  `title`s are allowed on anything, and [do technically become the name in this case](https://www.w3.org/TR/html-aam-1.0/#section-and-grouping-element-accessible-name-computation), but I don't think any accessibility technology would expect or use it as such here so I don't think we need to test it

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx (hunk 3 above)
Comment (cindyko226) [reply lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx]: Oh yes, updated here 5dc2262b93ca132993d670e04a39072e35273ad6, sorry for the confusion.

File: lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx (hunk 5 above)
Comment (cindyko226) [reply lib/src/components/NxSmallThreatCounter/__tests__/NxSmallThreatCounter.test.tsx]: Agree, updated here dbcb6ce377f41ca58742dba1a29eedf32f170992. Thanks.

**Review by rpokorny — APPROVED** [review]

**Pull request merged**